src/main/java/com/mraof/minestuck/blockentity/machine/AnthvilBlockEntity.java

```diff
@@ -0,0 +1,259 @@
+package com.mraof.minestuck.blockentity.machine;
+
+import com.mraof.minestuck.alchemy.*;
+import com.mraof.minestuck.alchemy.recipe.GristCostRecipe;
+import com.mraof.minestuck.blockentity.MSBlockEntityTypes;
+import com.mraof.minestuck.inventory.*;
+import com.mraof.minestuck.player.GristCache;
+import com.mraof.minestuck.util.ExtraForgeTags;
+import net.minecraft.core.BlockPos;
+import net.minecraft.core.Direction;
+import net.minecraft.nbt.CompoundTag;
+import net.minecraft.network.chat.Component;
+import net.minecraft.server.level.ServerPlayer;
+import net.minecraft.world.MenuProvider;
+import net.minecraft.world.entity.player.Inventory;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.inventory.AbstractContainerMenu;
+import net.minecraft.world.inventory.ContainerLevelAccess;
+import net.minecraft.world.inventory.DataSlot;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.level.block.state.BlockState;
+import net.minecraftforge.common.capabilities.Capability;
+import net.minecraftforge.common.capabilities.ForgeCapabilities;
+import net.minecraftforge.common.util.LazyOptional;
+import net.minecraftforge.items.IItemHandler;
+import net.minecraftforge.items.ItemStackHandler;
+import net.minecraftforge.items.wrapper.RangedWrapper;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.List;
+
+/**
+ * Mends an item at the cost of uranium power and grist. The grist used is the one with the highest value impact (weighted against build grist)
+ */
+public class AnthvilBlockEntity extends MachineProcessBlockEntity implements MenuProvider, UraniumPowered
+{
+	public static final String TITLE = "container.minestuck.anthvil";
+	public static final short MAX_FUEL = 128;
+	public static final short MEND_FUEL_COST = 5;
+	
+	private final ProgressTracker progressTracker = new ProgressTracker(ProgressTracker.RunType.ONCE, 0, this::setChanged, this::contentsValid);
```

Comment:
Using a progress tracker seems like overkill when it doesn't need to track progress and doesn't need to track if it's looping. Perhaps you could simplify it out?

Reply:
I kept it in under the uncritical assumption that it was necessary for classes extending MachineProcessBlockEntity. When I messed with the progress tracker before it prevented the GUI from showing up. Do you have any suggestions or examples of how to simplify it?

Reply:
You could cut out its usage bit by bit. First cut out use of the `MachinePacket.SetRunning` packet and trigger the refuel as part of `AnthvilPacket` and `AnthvilBlockEntity.attemptMend()` instead of through the progress tracker. After that, you could change `AnthvilMenu` to not require the progress tracker data by making it independent from `MachineContainerMenu` (that does mean that you need to handle the `ContainerLevelAccess` directly in `AnthvilMenu`, but that's no big deal. Once it's no longer used to trigger the refuel, and no longer synced through the menu, you can just remove it. You should be able to remove the ticker for the block entity at that point as well.

Does that seem reasonable with how refueling should happen?

Reply:
Have done so now!